plugins/outputs/amqp/amqp.go

```diff
@@ -249,7 +249,7 @@ func (q *AMQP) Write(metrics []telegraf.Metric) error {
 
 	if q.sentMessages >= q.MaxMessages && q.MaxMessages > 0 {
 		log.Printf("D! Output [amqp] sent MaxMessages; closing connection")
-		q.client = nil
+		q.client.Close()
```

Comment:
I think we need to set q.client = nil afterwards since this triggers a new connection in publish.